scripts/init.groovy.d/aab_apply_autoconfig_job.groovy

```diff
@@ -0,0 +1,32 @@
+// output init hook stage
+println '### JENKINS AUTOCONFIG: CREATING SELF-BOOTSTRAP JOB ###'
+
+/**
+  * Check if Jenkins should not configure itself and skip.
+  */
+
+Boolean skipConfiguringJenkins() {
+    !(System.env?.ENVIRONMENT in ['dev', 'staging']) ||
+        (new File("${Jenkins.instance.rootDir}/autoConfigComplete").exists())
+}
+if(skipConfiguringJenkins()) {
+    println 'Skipping Jenkins auto-configuration'
+    return
+}
+
+/**
+  * From this point onward Jenkins should configure itself.
+  */
+
+import jenkins.model.Jenkins
+import groovy.xml.*
```

Comment:
Suggestion (nit):
* Import necessary XML classes individually, only those that are required.


For example:

```
import groovy.xml.XMLParser
import groovy.xml.XMLSlurper  // if necessary — looks like we're only parsing here though
```

This feedback is non-blocking.